Dockerfile

```diff
@@ -0,0 +1,137 @@
+WORKDIR /
+RUN --mount=type=cache,target=/var/cache/apk,sharing=locked,id=apk \
+    apk add --no-cache git
+RUN git clone https://github.com/nikic/php-ast.git
+
+
+# Stage that builds the AST extension from downloaded src
+FROM builder-base AS ast-builder
+WORKDIR /build
+COPY --from=ast-downloader /php-ast .
+RUN phpize  \
+    && ./configure  \
+    && make  \
+    && make install
+RUN docker-php-ext-enable ast
+
+
+# Stage that builds the pcov extension from PECL
+FROM builder-base AS pcov-builder
+RUN pecl install pcov && docker-php-ext-enable pcov
+
+
+# Stage that builds all other extensions
+FROM builder-base AS extensions-builder
+RUN docker-php-ext-install bcmath intl zip pcntl bz2
+
+
+# Stage containing the downloaded toolbox PHAR
+FROM --platform=${BUILDPLATFORM} alpine AS toolbox-downloader
+WORKDIR /
+ARG TOOLBOX_VERSION
+RUN wget https://github.com/jakzal/toolbox/releases/download/v$TOOLBOX_VERSION/toolbox.phar
+
+
+# PHP with all extensions needed, correctly configured
+FROM php-base as php-configured
+
+# Extensions .so
+COPY --link --from=ast-builder /usr/local/lib/php/extensions /usr/local/lib/php/extensions
+COPY --link --from=pcov-builder /usr/local/lib/php/extensions /usr/local/lib/php/extensions
+COPY --link --from=extensions-builder /usr/local/lib/php/extensions /usr/local/lib/php/extensions
+
+# Extensions .ini
+COPY --link --from=ast-builder /usr/local/etc/php/conf.d /usr/local/etc/php/conf.d
+COPY --link --from=pcov-builder /usr/local/etc/php/conf.d  /usr/local/etc/php/conf.d
+COPY --link --from=extensions-builder /usr/local/etc/php/conf.d /usr/local/etc/php/conf.d
+
+COPY --link <<EOF /usr/local/etc/php/conf.d/jakzal.ini
+date.timezone=Europe/London
+memory_limit=-1
+phar.readonly=0
+pcov.enabled=0
+EOF
+
+COPY --link --chmod=755 <<EOF /entrypoint.sh
+#!/usr/bin/env sh
+set -e
+
+(test "\$GITHUB_ACTIONS" = "true" || test "\$CI" = "true") && test -f composer.json && composer install --no-scripts --no-progress
+
+exec "\$@"
+EOF
+ARG TOOLBOX_TARGET_DIR="/tools"
+COPY --link --from=toolbox-downloader --chmod=+x /toolbox.phar ${TOOLBOX_TARGET_DIR}/toolbox
+COPY --link --from=composer:2 /usr/bin/composer ${TOOLBOX_TARGET_DIR}/composer
+
+# INSTALLATION_DATE is used to bust the docker layer cache
+ARG INSTALLATION_DATE
+ARG TOOLBOX_EXCLUDED_TAGS
+ARG TOOLBOX_VERSION
+ENV TOOLBOX_EXCLUDED_TAGS=${TOOLBOX_EXCLUDED_TAGS}
+ENV TOOLBOX_TARGET_DIR="/tools"
+ENV TOOLBOX_VERSION=${TOOLBOX_VERSION}
+ENV COMPOSER_ALLOW_SUPERUSER 1
+ENV COMPOSER_HOME=$TOOLBOX_TARGET_DIR/.composer
+ENV PATH="$PATH:$TOOLBOX_TARGET_DIR:${COMPOSER_HOME}/vendor/bin:$TOOLBOX_TARGET_DIR/QualityAnalyzer/bin:$TOOLBOX_TARGET_DIR/DesignPatternDetector/bin:$TOOLBOX_TARGET_DIR/EasyCodingStandard/bin"
+
+RUN --mount=type=secret,id=composer.auth,target=${COMPOSER_HOME}/auth.json \
+    --mount=type=cache,id=composer,target=${COMPOSER_HOME}/cache\
+    --mount=type=secret,id=phive.auth,target=${TOOLBOX_TARGET_DIR}/.phive/auth.xml \
```

Comment:
In future for testing we need a local solution for this too

Reply:
What’s the problem with using this locally?

Reply:
Nothing bad happens; because the secret mounts aren't `required=true` they'll be ignored if not provided

It's more that we don't have anything in the Makefile that provides them

Reply:
e.g. (`--secret "type=file,id=composer.auth,source=~/.composer/auth.json` but fancier)